mitodl/ocw-studio
# Handle authentication for Concourse 7.7

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1112

#### What's this PR do?
Overrides the `concoursepy` auth function to handle the different responses to `../sky/login` returned by Concourse 7.7, while still maintaining backwards compatibility with Concourse 7.6.

#### How should this be manually tested?
- Try the following on your local concourse (`docker-compose --profile concourse up`), concourse CI (ask for credentials), and concourse RC:
- Create a new site and verify that a pipeline is created for it in concourse
- Publish the pipeline and verify that it triggers (don't worry about it failing).


#### Any background context you want to provide?
I forked the original `concoursepy` github repo to mitodl, but decided against using that fork here for now.  Just overriding the auth function is simplest to solve this problem at present.  If we do decide on using the fork now or later, we should probably also add it to pypi, set up doof to know about it and version it properly, update the version number in ocw-studio requirements, etc.
